cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -135,34 +136,56 @@ func (scaleSet *ScaleSet) GetScaleSetSize() (int64, error) {
 	return scaleSet.getCurSize()
 }
 
-// SetScaleSetSize sets ScaleSet size.
-func (scaleSet *ScaleSet) SetScaleSetSize(size int64) error {
-	scaleSet.mutex.Lock()
-	defer scaleSet.mutex.Unlock()
+// updateVMSSCapacity invokes virtualMachineScaleSetsClient to update the capacity for VMSS.
+func (scaleSet *ScaleSet) updateVMSSCapacity(size int64) {
+	var op compute.VirtualMachineScaleSet
+	var resp *http.Response
+	var isSuccess bool
+	var err error
+
+	defer func() {
+		if err != nil {
+			klog.Errorf("Failed to update the capacity for vmss %s with error %v, invalidate the cache so as to get the real size from API", scaleSet.Name, err)
+			// Invalidate the vmss size cache, so that it would be got by API.
+			scaleSet.mutex.Lock()
+			defer scaleSet.mutex.Unlock()
+			scaleSet.lastRefresh = time.Now().Add(-1 * 15 * time.Second)
```

Comment:
can you explain why we are setting last refresh to -15 seconds from Now()? Also, why not use Sub()? https://golang.org/pkg/time/#Time.Sub

Reply:
15 second is from L122, I renamed it to vmssSizeRefreshPeriod in PR #2151, but forgot to change here. Let me change it also to `vmssSizeRefreshPeriod` for clear.

Sub is a different use case, it accepts a param with `time.Time`, not `time.Duration`.